### ISSUE TITLE
Cannot change language - Uncaught TypeError: Argument 2 passed to Fisharebest\Webtrees\Statistics

### ISSUE THREAD

**Issue by magicsunday** [fisharebest/webtrees#2167]
I you try to change the language on https://dev.webtrees.net/demo-dev/index.php?route=tree-page&ged=demo

Uncaught TypeError: Argument 2 passed to Fisharebest\Webtrees\Statistics::__construct() must be an instance of Fisharebest\Webtrees\Tree, null given in /home/webtrees/dev.webtrees.net/demo-dev/app/Statistics.php:198
Stack trace:
#0 [internal function]: Fisharebest\Webtrees\Statistics-&gt;__construct(Object(Fisharebest\Webtrees\Services\ModuleService), NULL)
#1 /home/webtrees/dev.webtrees.net/demo-dev/vendor/illuminate/container/Container.php(812): ReflectionClass-&gt;newInstanceArgs(Array)
#2 /home/webtrees/dev.webtrees.net/demo-dev/vendor/illuminate/container/Container.php(658): Illuminate\Container\Container-&gt;build('Fisharebest\\Web...')
#3 /home/webtrees/dev.webtrees.net/demo-dev/vendor/illuminate/container/Container.php(609): Illuminate\Container\Container-&gt;resolve('Fisharebest\\Web...', Array)
#4 /home/webtrees/dev.webtrees.net/demo-dev/vendor/illuminate/container/Container.php(912): Illuminate\Container\Container-&gt;make('Fisharebest\\Web...')
#5 /home/webtrees/dev.webtrees.net/demo-dev/vendor/illuminate/contai in <b>/home/webtrees/dev.webtrees.net/demo-dev/app/Statistics.php</b> on line <b>198</b><br />

**Comment by magicsunday** [comment]
The "ged" parameter seems missing from the request.

**Comment by fisharebest** [comment]
As well as the `Null` trees and users, webtrees also has the concept of `Default` trees and users.  These have an ID of `-1`, and are used to store default settings.  For example the default blocks for new users.

So, being able to use `new DefaultTree()` and `new DefaultUser()` may be useful.

The `NullUser` object already exists - but is combined into the `User` class.  For example, if the `id` is zero, then preferences are saved in cookies rather that the database table.  Splitting these out might be useful.

We'd have:

* `UserInterface` with three implementations - `User`, `DefaultUser`, `GuestUser`.
* `TreeInterface`, with three implementations - `Tree`, `DefaultTree`, `NullTree`.

There's also places where we use a "null individual".

**Comment by fisharebest** [comment]
As a temporary fix, I have disabled the theme initialization for POST requests.

POST requests do not need themes - which means we do not need to load the modules.

**Comment by fisharebest** [comment]
There is also a 4th type of user object.  We have a "TreeUser", which is used to send emails.

**Comment by fisharebest** [comment]
I have implemented the `UserInterface` contract with various implementations `User`, `TreeUser`, `GuestUser`, `DefaultUser`, and moved functions for creating and authenticating users to a `UserService`.  This separation will be helpful when we implement other authentication methods.

I also refactored the module initialisation so that they aren't dependant on there being a current tree - and added this to the unit tests.
